static/js/components/ShiftCalendar.jsx

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Calendar, momentLocalizer, Views } from "react-big-calendar";
+import moment from "moment";
+import "react-big-calendar/lib/css/react-big-calendar.css";
+import { showNotification } from "baselayer/components/Notifications";
+import PropTypes from "prop-types";
+import * as shiftActions from "../ducks/shift";
+
+/* eslint-disable react/prop-types */
+
+const allViews = Object.keys(Views).map((k) => Views[k]);
+let dispatch;
+let currentUser;
+const localizer = momentLocalizer(moment);
+let groups;
+
+async function handleSelectSlot({ start, end }) {
+  const name = window.prompt("New Shift name");
+  const description = window.prompt("New Shift description");
+  const group_ids = groups.map((group) => `   ${group.name}: ${group.id}\n`);
+  const group_id = await window.prompt(
+    `Choose shift group ID : \n ${group_ids}`
+  );
+  const start_date = start.toISOString().replace("Z", "");
+  const end_date = end.toISOString().replace("Z", "");
+  dispatch(
+    shiftActions.submitShift({
+      name,
+      description,
+      start_date,
+      end_date,
+      group_id,
+    })
+  ).then((result) => {
+    if (result.status === "success") {
+      showNotification("Shift saved");
+    }
```

Comment:
Why do we have this notification in two different components?

Reply:
In the NewShift component, this notification triggers when you submit the shift you entered in the Form. In this component, it triggers when you create the Shift directly in the calendar, not with the form.